inductiva/resources/machines.py

```diff
@@ -17,73 +19,67 @@ def __init__(
         spot: bool = False,
         disk_size_gb: int = 20,
         zone: typing.Optional[str] = "europe-west1-b",
+        name: typing.Optional[str] = None,
     ) -> None:
         """Create a MachineGroup object.
 
         Args:
             machine_type: The type of GC machine to launch. Ex: "e2-medium".
-            Check https://cloud.google.com/compute/docs/machine-resource for 
+            Check https://cloud.google.com/compute/docs/machine-resource for
             more information about machine types.
             num_machines: The number of virtual machines to launch.
             spot: Whether to use spot machines.
             disk_size_gb: The size of the disk in GB, recommended min. is 20 GB.
             zone: The zone where the machines will be launched.
         """
         self.id = None
+        self.name = name
```

Comment:
If the user passes it, then it needs to be used to create the instance...
Otherwise, it is unintuitive. What we could do is call it label/tag and be a way to identify the MachineGroup but not its ID.

Problems that may surge are related with the unique ID. Since, if they create two machine groups with the same tag, we need to tell them they have two and they need to select which one they want.